Backpack/CardButton/Classes/SaveCardButton.swift

```diff
@@ -0,0 +1,254 @@
+
+                NSLayoutConstraint.activate([
+                    containedBackgroundCircle.widthAnchor.constraint(equalToConstant: circleSize),
+                    containedBackgroundCircle.heightAnchor.constraint(equalToConstant: circleSize),
+                    containedBackgroundCircle.centerXAnchor.constraint(equalTo: centerXAnchor),
+                    containedBackgroundCircle.centerYAnchor.constraint(equalTo: centerYAnchor)
+                ])
+                imageView.map { bringSubviewToFront($0) }
+            }
+        } else {
+            if containedBackgroundCircle.superview != nil {
+                containedBackgroundCircle.removeFromSuperview()
+            }
+        }
+    }
+
+    public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        updateLookAndFeel()
+    }
+}
+
+extension BPKSaveCardButton {
+    public func setChecked(_ newValue: Bool, animated: Bool) {
+        guard animated, let imageView = imageView else {
+            checked = newValue
+            return
+        }
+        if newValue {
+            // Update icon color before animation starts
+            checked = true
+            // 1. resize to 30px
+            // 2. wait
+            // 3. resize back to 24px
+            let transformBefore = imageView.transform
+            UIView.animate(
+                withDuration: BPKDuration.animationDurationSm,
+                delay: 0,
+                options: UIView.AnimationOptions([.beginFromCurrentState, .curveEaseOut])
+            ) {
+                imageView.transform = CGAffineTransformMakeScale(1.25, 1.25)
+            }
+            UIView.animate(
+                withDuration: BPKDuration.animationDurationSm,
+                delay: BPKDuration.animationDurationSm * 2,
+                options: UIView.AnimationOptions([.beginFromCurrentState, .curveEaseIn])
+            ) {
+                imageView.transform = transformBefore
+            }
+        } else {
+            // return to default state with transition
+            UIView.transition(
+                with: self,
+                duration: BPKDuration.animationDurationSm,
+                options: .transitionCrossDissolve
+            ) {
+                self.checked = newValue
+            }
+        }
+    }
+}
+
+private extension BPKSaveCardButton {
+
+    private struct Icons {
+        let smallIcon: BPKSmallIconName
+        let smallIconHighlighted: BPKSmallIconName
+        let defaultIcon: BPKLargeIconName
+        let defaultIconHighlighted: BPKLargeIconName
+    }
+
+    private struct Colors {
+        let iconColorOnDark: UIColor
+        let iconColorContained: UIColor
+        let iconColorDefault: UIColor
+        let highlightedIconColorOnDark: UIColor
+        let highlightedIconColorContained: UIColor
+        let highlightedIconColorDefault: UIColor
```

Comment:
These would preferably be BPKColor

Reply:
BPKColor.textOnDarkColor return a UIColor in Backpack. 
Should I import Backpack-SwiftUI's BPKDynamicColors?

Reply:
No, this is fine then :D